utils/middleware/error-handling-middleware.ts

```diff
@@ -3,15 +3,23 @@ import { APIGatewayProxyResult } from "aws-lambda";
 import { serializeError } from "serialize-error";
 import { ValidationErrorResponse } from "../response";
 import { INTERNAL_SERVER_ERROR, REQUEST_BODY_INVALID } from "../errors";
-import { ILambdaEvent } from "../../models/provisioning-jobs";
+import { StepFunctionRequestEvent, RequestBodyType, CspInvocation, CspResponse } from "../../models/provisioning-jobs";
 
-export const errorHandlingMiddleware = (): middy.MiddlewareObj<ILambdaEvent, APIGatewayProxyResult> => {
-  const onError: middy.MiddlewareFn<ILambdaEvent, APIGatewayProxyResult> = async (
-    request
-  ): Promise<ValidationErrorResponse | void> => {
-    const error = serializeError(request.error)!;
+export const errorHandlingMiddleware = (): middy.MiddlewareObj<
+  StepFunctionRequestEvent<RequestBodyType> | CspInvocation,
+  APIGatewayProxyResult | CspResponse | ValidationErrorResponse
```

Comment:
This feels like a lot of possible types. Can you help me understand what's going on here?

Reply:
@kylelaker-ccpo This comes from the different inputs and outputs being used for the `baseHandler`s. The `start-provision-job` lambda is initiated from a `POST` request, which output is sent to the step function `mock-invocation-lambda` and is of a different shape. The various types allows for the middlewares to handle the different inputs and outputs without throwing a typescript error since it is based on the inputs and outputs being used by the `baseHandler`.
- `StepFunctionRequestEvent<RequestBodyType> | CspInvocation` are the allowed inputs 
- `APIGatewayProxyResult | CspResponse | ValidationErrorResponse` are the allowed outputs

I guess a better way might be something like 
```typescript
type middlwareInputs = StepFunctionRequestEvent<RequestBodyType> | CspInvocation
type middlewareOutputs = APIGatewayProxyResult | CspResponse | ValidationErrorResponse

export const errorHandlingMiddleware = (): middy.MiddlewareObj<middlewareInputs, middlewareOutputs>
```

Reply:
Oh oops, I missed the comma; sorry! But yeah; that would be a lot cleaner! Maybe with semicolon?

Reply:
No problem. Oh yeah semicolon for sure